client/blocks/stripe-utils/normalize.js

```diff
@@ -0,0 +1,176 @@
+	const rates = shippingOptions[ 0 ].shipping_rates;
+	return rates.map( ( rate ) => {
+		return {
+			id: rate.rate_id,
+			label: rate.name,
+			detail: rate.description,
+			amount: parseInt( rate.price, 10 ),
+		};
+	} );
+};
+
+/**
+ * Normalize shipping address information from stripe's address object to
+ * the cart shipping address object shape.
+ *
+ * @param {StripeShippingAddress} shippingAddress Stripe's shipping address item
+ *
+ * @return {CartShippingAddress} The shipping address in the shape expected by
+ * the cart.
+ */
+const normalizeShippingAddressForCheckout = ( shippingAddress ) => {
+	const address = {
+		first_name: shippingAddress.recipient
+			.split( ' ' )
+			.slice( 0, 1 )
+			.join( ' ' ),
+		last_name: shippingAddress.recipient
+			.split( ' ' )
+			.slice( 1 )
+			.join( ' ' ),
+		company: '',
+		address_1:
+			typeof shippingAddress.addressLine[ 0 ] === 'undefined'
+				? ''
+				: shippingAddress.addressLine[ 0 ],
+		address_2:
+			typeof shippingAddress.addressLine[ 1 ] === 'undefined'
+				? ''
+				: shippingAddress.addressLine[ 1 ],
+		city: shippingAddress.city,
+		state: shippingAddress.region,
+		country: shippingAddress.country,
+		postcode: shippingAddress.postalCode.replace( ' ', '' ),
+	};
+	return address;
+};
+
+/**
+ * Normalizes shipping option shape selection from Stripe's shipping option
+ * object to the expected shape for cart shipping option selections.
+ *
+ * @param {StripeShippingOption} shippingOption The customer's selected shipping
+ *                                              option.
+ *
+ * @return {string[]}  An array of ids (in this case will just be one)
+ */
+const normalizeShippingOptionSelectionsForCheckout = ( shippingOption ) => {
+	return shippingOption.id;
+};
+
+/**
+ * Returns the billing data extracted from the stripe payment response to the
+ * CartBillingData shape.
+ *
+ * @param {StripePaymentResponse} paymentResponse Stripe's payment response
+ *                                                object.
+ *
+ * @return {CartBillingAddress} The cart billing data
+ */
+const getBillingData = ( paymentResponse ) => {
+	const source = paymentResponse.source;
+	const name = source && source.owner.name;
+	const billing = source && source.owner.address;
+	const payerEmail = paymentResponse.payerEmail || '';
+	const payerPhone = paymentResponse.payerPhone || '';
+	return {
+		first_name: name ? name.split( ' ' ).slice( 0, 1 ).join( ' ' ) : '',
+		last_name: name ? name.split( ' ' ).slice( 1 ).join( ' ' ) : '',
+		email: ( source && source.owner.email ) || payerEmail,
+		phone:
+			( source && source.owner.phone ) ||
+			payerPhone.replace( '/[() -]/g', '' ),
+		country: ( billing && billing.country ) || '',
+		address_1: ( billing && billing.line1 ) || '',
+		address_2: ( billing && billing.line2 ) || '',
+		city: ( billing && billing.city ) || '',
+		state: ( billing && billing.state ) || '',
+		postcode: ( billing && billing.postal_code ) || '',
+		company: '',
```

Comment:
Let's change the code here so that it uses [null coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) and [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

Reply:
Fixed in d0afa3a79374e886e14b8946d147bfa509812051